### PR TITLE
Docs: move custom tranport guide to the new location

### LINKED ISSUE #3990: Docs: Topics (transport plugins)

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #3990

### CODE REVIEW

**Comment by sphuber** [comment]
Just a heads up @yakutovicha : as noted in the Slack, all plugin writing guides should go into "Topics" and not a How-to

**Comment by yakutovicha** [comment]
The documentation looks good to me. I read the text, checked the links and tried to create a very simple plugin from the template provided in the documentation. All seems to work fine.

File: docs/source/topics/transport.rst [hunk 1]
```diff
@@ -1,5 +1,8 @@
+.. _how-to:transport:
```
Comment (sphuber) [inline docs/source/topics/transport.rst]:
```suggestion
.. _topics:transport:
```

File: docs/source/topics/transport.rst [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+
+*****************
 Transport plugins
-=================
+*****************
 
 .. toctree::
    :maxdepth: 2
```
Comment (sphuber) [inline docs/source/topics/transport.rst:8]: Is this toctree really necessary? There is not much content here, right?

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @yakutovicha . In addition to my inline comments, you also need to add proper code block statement to line 32, so it should become:
```
.. code-block:: python

    with TransportPlugin() as transport:
        transport.some_method()
```

File: docs/source/topics/transport.rst (hunk 1 above)
Comment (yakutovicha) [reply docs/source/topics/transport.rst]: sorry, already done :)

**Review by giovannipizzi — DISMISSED** [review]

**Review by sphuber — CHANGES_REQUESTED** [review]
Please add a redirect rule to `docs/source/redirects.txt`
```
developer_guide/core/transport.rst topics/transport.rst 
```

**Review by sphuber — APPROVED** [review]
Thanks @yakutovicha

**Pull request merged**